tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -340,62 +341,71 @@ def _enable_disable_volume_profiling(self):
             integration_id=NS.tendrl_context.integration_id
         ).load()
         volumes = NS.gluster.objects.Volume().load_all() or []
-        failed_vols = []
-        if cluster.volume_profiling_flag == "enable":
-            for volume in volumes:
-                if volume.profiling_enabled == "yes":
-                    continue
-                out, err, rc = cmd_utils.Command(
-                    "gluster volume profile %s start" %
-                    volume.name
-                ).run()
-                if (err or rc != 0) and \
-                    "already started" in err:
-                    failed_vols.append(volume.name)
-            if len(failed_vols) > 0:
-                logger.log(
-                    "debug",
-                    NS.publisher_id,
-                    {
-                        "message": "Profiling already "
-                        "enabled for volumes: %s" %
-                        str(failed_vols)
-                    }
-                )
-            cluster.volume_profiling_state = "enabled"
-        if cluster.volume_profiling_flag == "disable":
-            for volume in volumes:
-                if volume.profiling_enabled == "no":
-                    continue
-                out, err, rc = cmd_utils.Command(
-                    "gluster volume profile %s stop" %
-                    volume.name
-                ).run()
-                if (err or rc != 0) and \
-                    "not started" in err:
-                    failed_vols.append(volume.name)
-            if len(failed_vols) > 0:
-                logger.log(
-                    "debug",
-                    NS.publisher_id,
-                    {
-                        "message": "Profiling not "
-                        "enabled for volumes: %s" %
-                        str(failed_vols)
-                    }
-                )
-            cluster.volume_profiling_state = "disabled"
-        if cluster.volume_profiling_flag == "leave-as-is":
-            profiling_enabled_count = 0
-            for volume in volumes:
-                if volume.profiling_enabled == "yes":
-                    profiling_enabled_count += 1
-            if profiling_enabled_count == 0:
-                cluster.volume_profiling_state = "disabled"
-            elif profiling_enabled_count == len(volumes):
+        # Enable / disable based on cluster flag volume_profiling_flag
+        # should be done only once while first sync. Later the volume
+        # level volume_profiling_state should be set based on individual
+        # volume level values
+        first_sync_done = etcd_utils.read(
+            "/clusters/%s/nodes/%s/NodeContext/first_sync_done" %
+            (NS.tendrl_context.integration_id,
+             NS.node_context.node_id)
+        ).value
```

Comment:
Why direct read, use NS.tendrl.objects.ClusterNodeContext()

Reply:
I remember I faced some issues while reading `first_sync_done` field. And as a quick solution I did this. Will check that.